qiskit/transpiler/passes/layout/sabre_layout.py

```diff
@@ -244,20 +242,14 @@ def run(self, dag):
         ) in layout_components:
             initial_layout_dict.update({k: component_map[v] for k, v in layout_dict.items()})
             final_layout_dict.update({component_map[k]: component_map[v] for k, v in final_dict})
-            if not shared_clbits:
-                for clbit in local_dag.clbits:
-                    if clbit in seen_clbits:
-                        shared_clbits = True
-                        break
-                    seen_clbits.add(clbit)
         self.property_set["layout"] = Layout(initial_layout_dict)
         # If skip_routing is set then return the layout in the property set
         # and throwaway the extra work we did to compute the swap map.
-        # We also skip routing here if the input circuit is split over multiple
-        # connected components and there is a shared clbit between any
-        # components. We can only reliably route the full dag if there is any
-        # shared classical data.
-        if self.skip_routing or shared_clbits:
+        # We also skip routing here if there is more than one connected
+        # component we ran layout on. We can only reliably route the full dag
+        # in this case if there is any dependency between the components
+        # (typically shared classical data or barriers).
```

Comment:
This is a bit unclear to me. What does this mean ?
> there is more than one connected  component we ran layout on
There are more than one connected components. Did we run layout on some of them?
Can we reliably route if there is a dependency or if there is no dependency?

Reply:
The way the layout problem gets split up is a function of both the circuit and the coupling map. If the coupling map is disjoint and the circuit isn't we only run layout over a single connected component from the coupling map. The only case when we're not safe to execute routing as part of this pass is when we have more than one circuit component and coupling map component so that we run the inner layout function >1 time. In those cases there is a potential data dependency across the output from the split layout runs' connected components (which is the combination of the connected components from the circuit applied to connected components on the device) in the non-quantum component of the circuit.